api_grpc/worker.go

```diff
@@ -440,6 +443,49 @@ func NotifyWorkers() {
 	}
 }
 
+func NotifyWorkerToStopStream(stream model.Stream) {
+	workers := workersForStream[stream.ID]
+	if len(workers) == 0 {
+		log.Error("Can't get lecture hall for stream")
+		return
+	}
+
+	// Iterate over all workers that are used for the given stream
+	for _, currentWorker := range workers {
+		credentials := insecure.NewCredentials()
+
+		// Connect to the worker and tell it to end the stream
+		log.Error("Connecting to:" + fmt.Sprintf("%s:50051", currentWorker.Host))
+		conn, err := grpc.Dial(fmt.Sprintf("%s:50051", currentWorker.Host), grpc.WithTransportCredentials(credentials))
+		if err != nil {
+			log.WithError(err).Error("Unable to dial server")
+			_ = conn.Close()
+			return
+		}
+
+		client := pb.NewToWorkerClient(conn)
+
+		req := pb.EndStreamRequest{
+			StreamID: uint32(stream.ID),
+			WorkerID: currentWorker.WorkerID,
+		}
+		resp, err := client.RequestStreamEnd(context.Background(), &req)
+
+		// Was ending the stream successful?
+		if err != nil || !resp.Ok {
+			log.WithError(err).Error("Could not end stream!")
+			_ = conn.Close()
+			return
+		}
+
+		_ = conn.Close()
+
+	}
+	// All workers for stream are assumed to done
+	// TODO: Remove individual workers in NotifyStreamFinished?
```

Comment:
Assuming all are done is just fine :) But again, this should happen in the database.